src/commonMain/kotlin/org/cru/godtools/tool/xml/XmlPullParser.kt

```diff
@@ -0,0 +1,36 @@
+package org.cru.godtools.tool.xml
+
+internal interface XmlPullParser {
+    val eventType: Int
+    val namespace: String?
+    val name: String?
+
+    fun require(type: Int, namespace: String? = null, name: String? = null)
+
+    fun next(): Int
+    fun nextTag(): Int
+    fun nextText(): String
+
+    fun getAttributeValue(namespace: String?, name: String): String?
+
+    companion object {
+        const val START_DOCUMENT = 0
+        const val END_DOCUMENT = 1
+        const val START_TAG = 2
+        const val END_TAG = 3
+        const val TEXT = 4
+    }
+}
+
+internal fun XmlPullParser.skipTag() {
+    require(XmlPullParser.START_TAG, null, null)
+
+    // loop until we process all nested tags
+    var depth = 1
+    while (depth > 0) {
+        when (next()) {
+            XmlPullParser.START_TAG -> depth++
+            XmlPullParser.END_TAG -> depth--
+        }
+    }
```

Comment:
So how does this fully stop? Maybe I'm misunderstanding the logic or missing something. If depth starts at 1 and then goes to 2 when it hits the START TAG, I assume when it gets to the END_TAG we would want the loop to stop. But when it hits the END_TAG, depth is currently at 2 and would just go down to 1, so the loop would continue right? Unless the ```skipTag``` method above helps with this?

Reply:
```
<root>
  <child></child>
</root>
```

this XML would produce the following stream of events:
1. `START_TAG` - root
2. `START_TAG` - child
3. `END_TAG` - child
4. `END_TAG` - root

when we enter this function, the current event is 1. the switch on line 31 will advance to event 2 before evaluating the condition. You should be able to follow the stream after that.

Basically we `require` this method be called when the current event is a `START_TAG`, then consume all events until we encounter the matching `END_TAG`

Reply:
Ah the snippet you added helps, thanks!